### ISSUE TITLE
[Bug] GZCGQ01LM Not Supported in Apple Home after update z2m to new 2.0 scheme

### ISSUE THREAD

**Issue by mrmaximas** [itavero/homebridge-z2m#966]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

After latest dev z2m update GZCGQ01LM not supported in Apple Home
I think this appear after illuminance_lux renamed to illuminance according to this https://github.com/Koenkk/zigbee2mqtt/discussions/24198#discussion-7273807

### Related devices

GZCGQ01LM
Also affected my RTCGQ14LM and RTCGQ11LM - light sensor has disappeared

### Steps To Reproduce

Update z2m to latest dev

### This plugin

1.11.0-beta.6

### Homebridge

1.8.5

### Zigbee2MQTT

1.42.0-dev commit: 639e3bb6

**Comment by mrmaximas** [comment]
@itavero please pay attention. thanks!

**Comment by itavero** [comment]
> @itavero please pay attention. thanks!

Please don't be rude. I'll have a look when I have time. Considering the festivities in December, and my wife and I expecting our second child beginning of 2025, this might not be within 4 days after you reported an issue.

Also, considering that you are not my wife or mother, nor paying my bills, don't tell me what to do.

If it's not going fast enough, consider looking into the issue yourself and providing a PR to resolve it.

**TLDR:** Don't nag to maintainers of open source software. Be patient. I'll get to it when I get to it. This is definitely not the most important thing in my life.

**Comment by itavero** [comment]
Latest beta should solve this. Let me know if you still run into issues 😉

**Comment by mrmaximas** [comment]
@itavero thanks! 
But "bootloop" detected
1/5/2025, 12:17:35 AM] [homebridge-z2m] Connected to MQTT server
[1/5/2025, 12:17:35 AM] [homebridge-z2m] Zigbee2MQTT is ONLINE
[1/5/2025, 12:17:35 AM] [homebridge-z2m] Using Zigbee2MQTT v2.0.0-dev (identified via zigbee2mqtt/bridge/info)
[1/5/2025, 12:17:35 AM] [homebridge-z2m] Child bridge process ended
[1/5/2025, 12:17:35 AM] [homebridge-z2m] Process Ended. Code: 1, Signal: null

**Comment by itavero** [comment]
Hm.. didn't see that while trying it locally during development. Maybe a combination of things.. I'll have a look. Thanks for the quick feedback!

**Comment by mrmaximas** [comment]
`{
    "mqtt": {
        "base_topic": "zigbee2mqtt",
        "server": "mqtt://192.168.88.10:1883",
        "reject_unauthorized": false,
        "keepalive": 60,
        "version": 4,
        "disable_qos": false
    },
    "log": {
        "mqtt_publish": "debug"
    },
    "defaults": {
        "exclude": false,
        "excluded_keys": [
            "device_temperature"
        ]
    },
    "exclude_grouped_devices": true,
    "devices": [
        {
            "id": "0xa4c138ceebf27157",
            "exclude": true
        },
        {
            "id": "0x187a3efffe358554",
            "exclude": true
        },
        {
            "id": "0xa4c138510c951009",
            "exclude": true
        },
        {
            "id": "0xa4c138b70db602ea",
            "exclude": true
        },
        {
            "id": "0xa4c13807f04b19ac",
            "exclude": true
        },
        {
            "id": "0xa4c138b5bb325444",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 90
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c13896c5aa4ff2",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 30
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c13828adf5d72d",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 30
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c138243e41c8dd",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 90
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c1385c28385d70",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 30
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c1388b0b11bcbb",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 80
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c13889c384dbed",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 80
                        }
                    ],
                    "type": "light"
                }
            ]
        }
    ],
    "_bridge": {
        "username": "0E:7A:DD:3D:DA:BA",
        "port": 53072
    },
    "platform": "zigbee2mqtt"
}`

**Comment by mrmaximas** [comment]
{
    "mqtt": {
        "base_topic": "zigbee2mqtt",
        "server": "mqtt://192.168.188.110:1883",
        "reject_unauthorized": false,
        "keepalive": 60,
        "version": 4,
        "disable_qos": false
    },
    "log": {
        "mqtt_publish": "debug"
    },
    "defaults": {
        "exclude": false,
        "excluded_keys": [
            "device_temperature"
        ]
    },
    "exclude_grouped_devices": true,
    "devices": [
        {
            "id": "0xa4c138ceebf27157",
            "exclude": true
        },
        {
            "id": "0x187a3efffe358554",
            "exclude": true
        },
        {
            "id": "0xa4c138510c951009",
            "exclude": true
        },
        {
            "id": "0xa4c138b70db602ea",
            "exclude": true
        },
        {
            "id": "0xa4c13807f04b19ac",
            "exclude": true
        },
        {
            "id": "0xa4c138b5bb325444",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 90
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c13896c5aa4ff2",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 30
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c13828adf5d72d",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 30
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c138243e41c8dd",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 90
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c1385c28385d70",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 30
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c1388b0b11bcbb",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 80
                        }
                    ],
                    "type": "light"
                }
            ]
        },
        {
            "id": "0xa4c13889c384dbed",
            "exposes": [
                {
                    "features": [
                        {
                            "access": 7,
                            "label": "State",
                            "name": "state",
                            "property": "state",
                            "type": "binary",
                            "value_off": "OFF",
                            "value_on": "ON",
                            "value_toggle": "TOGGLE"
                        },
                        {
                            "access": 7,
                            "label": "Brightness",
                            "name": "brightness",
                            "property": "brightness",
                            "type": "numeric",
                            "value_max": 254,
                            "value_min": 80
                        }
                    ],
                    "type": "light"
                }
            ]
        }
    ],
    "_bridge": {
        "username": "0E:7A:DD:3D:DA:BA",
        "port": 53072
    },
    "platform": "zigbee2mqtt"
}

**Comment by mrmaximas** [comment]
[homebridge-z2m.log.txt](https://github.com/user-attachments/files/18308323/homebridge-z2m.log.txt)
please download the i remove it

**Comment by itavero** [comment]
Thanks. Found the exception. I'll have a look.

```
TypeError: Cannot use 'in' operator to search for 'state' in undefined
    at Zigbee2mqttPlatform.handleDeviceAvailability (/usr/local/lib/node_modules/homebridge-z2m/src/platform.ts:411:18)
    at Zigbee2mqttPlatform.onMessage (/usr/local/lib/node_modules/homebridge-z2m/src/platform.ts:296:14)
    at MqttClient.emit (node:events:524:28)
    at handlePublish (/usr/local/lib/node_modules/homebridge-z2m/node_modules/mqtt/src/lib/handlers/publish.ts:172:11)
    at handle (/usr/local/lib/node_modules/homebridge-z2m/node_modules/mqtt/src/lib/handlers/index.ts:31:17)
    at work (/usr/local/lib/node_modules/homebridge-z2m/node_modules/mqtt/src/lib/client.ts:779:17)
    at processTicksAndRejections (node:internal/process/task_queues:85:11)
```

**Comment by itavero** [comment]
@mrmaximas Would you be able to try the changes in #986 without me making a release already?

**Comment by mrmaximas** [comment]
@itavero Just updated to the latest beta. everything works ok! Thanks!
